src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,12 +1048,25 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static ReadOnlyCollection<SyntaxKind> _lazyContextualKeywordKindsArray;
```

Comment:
I added the `isnot` keyword and found this issue accidentially only. There is no comment (at least) in the keyword enum that tells me that this function has also to be updated. That gave me the idea to make this function independent of the keyword list. Chance is great that other context keywords will be added in any future release.